vllm/entrypoints/openai/serving_chat.py

```diff
@@ -515,8 +660,11 @@ async def chat_completion_full_generator(
             else:
                 logprobs = None
 
+            tools_called = False
+            # if the reqeust uses tools and specified a tool choice
             if request.tool_choice and type(
```

Comment:
There is a problem here, after this if ... elif .. elif block, the variable `message` might not be defined if we never encountered a valid case.

This results in an internal server error.

I would suggest to:
 * move the `elif not request.tool_choice or request.tool_choice == "none":` to an `else` clause at the end
 * or raise a proper exception
 * or add an else clause and set message to some valid default value.

Reply:
Thanks! I'm going to roll with logging an error and returning a non-parsed chat completion message to avoid broken things down the line. 

Please see 95605912